_episodes/11-dask-configuration.md

```diff
@@ -0,0 +1,390 @@
+compatibility: ESMValCore v2.10.0
+
+questions:
+  - "What is the Dask configuration file and how should I use it?"
+  - "What are Dask workers"
+  - "What is the Dask scheduler"
+
+objectives:
+  - "Understand the contents of the dask.yml file"
+  - "Prepare a personalized dask.yml file"
+
+keypoints:
+  - "The ``~/.esmvaltool/dask.yml`` file tells ESMValCore how to configure Dask."
+  - "``cluster`` can be used to start a new Dask cluster for each run."
+  - "``client`` can be used to connect to an already running Dask cluster."
+  - "The Dask default scheduler can be configured by editing the files in ``~/.config/dask``."
+  - "The Dask Dashboard can be used to see if the Dask workers have sufficient memory available."
+
+---
+
+## Introduction
+
+When processing larger amounts of data, and especially when the tool crashes
+when running a recipe because there is not enough memory available, it is
+usually beneficial to change the default [Dask configuration][dask-configuration].
+
+The preprocessor functions in ESMValCore use the
+[Iris](https://scitools-iris.readthedocs.io) library, which in turn uses Dask
+Arrays to be able to process datasets that are larger than the available memory.
+It is not necesary to understand how these work exactly to use the ESMValTool,
+but if you are interested there is a
+[Dask Array Tutorial](https://tutorial.dask.org/02_array.html) as a well as a
+[guide to "Lazy Data"](https://scitools-iris.readthedocs.io/
+en/stable/userguide/real_and_lazy_data.html)
+available. Lazy data is the term the Iris library uses for Dask Arrays.
+
+
+### Dask Workers
+The most important concept to understand when using Dask Arrays is the concept
+of a Dask *worker*. With Dask, computations are run in parallel by little
+Python programs that are called *workers*. These could be on running on the
+same machine that you are running ESMValTool on, or they could be on one or
+more other computers. Dask workers typically require 2 to 4 gigabytes (GiB) of
+memory (RAM) each. In order to avoid running out of memory, it is important
+to use only as many workers as your computer(s) have memory for. ESMValCore
+(or Dask) provide configuration files where you can configure the number of
+workers.
+
+Note that only array computations are run using Dask, so total runtime may not
+decrease as much as you might expect when you increase the number of Dask
+workers.
+
+### Dask Scheduler
+
+In order to distribute the computations over the workers, Dask makes use of a
+*scheduler*. There are two different schedulers available. The default
+scheduler can be a good choice for smaller computations that can run
+on a single computer, while the scheduler provided by the Dask Distributed
+package is more suitable for larger computations.
+
+> ## On using ``max_parallel_tasks``
+>
+> In the config-user.yml file, there is a setting called ``max_parallel_tasks``.
+> Any variable or diagnostic script in the recipe is considered a 'task' in this
```

Comment:
Suggestion: variable ->variable to be preprocessed.

Reply:
changed in [a74a1ab](https://github.com/ESMValGroup/ESMValTool_Tutorial/pull/332/commits/a74a1abd7aa6edf3bce2e89a677407f4cca0b53d)